src/target/kinetis.c

```diff
@@ -253,6 +286,58 @@ static int kl_gen_flash_done(struct target_flash *f)
 	return 0;
 }
 
+
+static int mk_gen_flash_erase(struct target_flash *f, target_addr addr, size_t len){
+	while (len) {
+		if (kl_gen_command(f->t, FTFA_CMD_ERASE_SECTOR, addr, NULL)) {
+			len -= MK_GEN_SECTOR_SIZE;
+			addr += MK_GEN_SECTOR_SIZE;
+		} else {
+			return 1;
+		}
+	}
+	return 0;
+}
+
+static int mk_gen_flash_write(struct target_flash *f,
+                              target_addr dest, const void *src, size_t len){
+	/* Ensure we don't write something horrible over the security byte */
+	if (!unsafe_enabled &&
+		(dest <= FLASH_SECURITY_BYTE_ADDRESS) &&
+		((dest + len) > FLASH_SECURITY_BYTE_ADDRESS)) {
+		((uint8_t*)src)[FLASH_SECURITY_BYTE_ADDRESS - dest] =
+			FLASH_SECURITY_BYTE_UNSECURED;
+	}
+
+	while (len) {
+		if (kl_gen_command(f->t, FTFE_CMD_PROGRAM_PHRASE, dest, src)) {
+			len -= 8;
+			dest += 8;
+			src += 8;
+		} else {
+			return 1;
+		}
+	}
+	return 0;
+}
+
+static void mk_gen_add_flash(target *t,
+                           uint32_t addr, size_t length, size_t erasesize){
+
+	struct target_flash *f = calloc(1, sizeof(*f));
+
+	f->start = addr;
+	f->length = length;
+	f->blocksize = erasesize;
+	f->erase = mk_gen_flash_erase;
+	f->write = mk_gen_flash_write;
+	f->done = kl_gen_flash_done;
```

Comment:
Is this going to work?  `kl_gen_flash_done` may use `FTFA_CMD_PROGRAM_LONGWORD` to write the security byte.